[jit][edge] Add control stack frame to lite interpreter

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65972
* #65971
* #65970
* #65969
* #65968
* #65967
* #65966
* #65965
* #65964
* __->__ #65963





Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

zhxchen17 committed:
[jit][edge] Add control stack frame to lite interpreter



Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"


Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

[ghstack-poisoned]

cccclai commented on torch/csrc/jit/mobile/interpreter.h:
```diff
@@ -29,13 +32,17 @@ struct Code {
 
 struct InterpreterState {
-  TORCH_API explicit InterpreterState(std::shared_ptr<Code> code);
+  TORCH_API explicit InterpreterState(const Code& code);
```
Any specific reason to change from `shared_ptr` to pass by reference here?

zhxchen17 replied:
InterpreterState doesn't outlive the Function calling it, so it doesn't have to own the Code. 
The general problems of std::shared_ptr are 1. Performance-wise it needs to increment/decrement atomic counters on their own. 2. More importantly interface-wise using them makes us leaking the shared ownership everywhere in codebase, making the future change harder.

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

cccclai reviewed (approved):
LGTM. Thanks!

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Add control stack frame to lite interpreter"






Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150/)

Differential Revision: [D31326150](https://our.internmc.facebook.com/intern/diff/D31326150)

[ghstack-poisoned]
